golang/cosmos/x/lien/lien.go

```diff
@@ -60,12 +77,77 @@ func (ch portHandler) Receive(ctx *vm.ControllerContext, str string) (string, er
 	case LIEN_GET_ACCOUNT_STATE:
 		return ch.handleGetAccountState(ctx.Context, msg)
 
+	case LIEN_GET_STAKING:
+		return ch.handleGetStaking(ctx.Context, msg)
+
 	case LIEN_SET_LIENED:
 		return ch.handleSetLiened(ctx.Context, msg)
 	}
 	return "", fmt.Errorf("unrecognized type %s", msg.Type)
 }
 
+func (ch portHandler) handleGetStaking(ctx sdk.Context, msg portMessage) (string, error) {
+	reply := msgStaking{
+		EpochTag:        fmt.Sprint(ctx.BlockHeight()),
+		Denom:           ch.keeper.BondDenom(ctx),
+		ValidatorValues: make([]sdk.Int, 0),
+		DelegatorStates: make([]delegatorState, 0),
```

Comment:
Should preallocate the slices:
```suggestion
		ValidatorValues: make([]sdk.Int, 0, len(msg.Validators)),
		DelegatorStates: make([]delegatorState, 0, len(msg.Delegators)),
```

Reply:
Done - simply allocating to final length since it's exact.